### ISSUE TITLE
Notifications EPIC

### ISSUE THREAD

**Issue by idnorton** [emfcamp/Website#870]
We need outbound notifications to volunteers in a variety of flavours. Some suggestions so far

* Email
* IRC
* Twitter
* SMS

This isn't to track the individual implementation tasks but to ask as a wrapper for subtasks implementing the specifics.

Thus far it's been requested that we send notifications to people registered for a shift 30 minutes before their shift to remind them. We need to:

* Support config of this in the volunteer facing UI
* Support for this in the volunteer admin UI

What other aspects of this would we like to consider?

Ad-hoc notifications to:

* individual or arbitrary selection of people
* a given shift
* people registered for a given role
* anything else?

**Comment by russss** [comment]
I'm going to add one more consideration here briefly:

The aim for this year is to eventually get the website operating as a progressive web app. This will allow people to install it as an app on their phone, and will also mean we can potentially send push notifications.

Given that the mobile coverage at Eastnor is not great, I think we should rule out SMS notifications. I recommend probably starting with email, then we can look at using push notifications later rather than SMS.

**Comment by idnorton** [comment]
@russss yup good call. Email is our MVP for this year, the rest are consideration at this point.

**Comment by Jonty** [comment]
FWIW When I mentioned SMS I was specifically talking about doing it via Twilio, thinking that we could also use their WhatsApp support as that would work fine on site.

Unsure if Signal/etc also have API's we could use, or if there's an API that aggregates them.

**Comment by idnorton** [comment]
Notes for the future from discussion with the web team on IRC:

* Email sending is queued in a PG table https://github.com/emfcamp/Website/blob/master/models/email.py
* At the moment we don't mark items completed and the table naming etc assumes email is our only notification type
* Rename the table to be more generic, add a column for notification type and also a bool so that we know it's been sent
* Be nice to add an additional table for logging actions in the event of retries etc so we know when to give up

The function `get_export_data` in the model is responsible for post-event export and will summarise stats on handling so this will need to be updated to cope with the changes mentioned above.

**Comment by russss** [comment]
There is a `sent` column in EmailJobRecipient, and there's a [background job](https://github.com/emfcamp/Website/blob/master/apps/base/scheduled_tasks.py) which runs every minute, sends the emails, and marks them as sent in the database.
